rust-runtime/aws-smithy-http-server/src/operation/mod.rs

```diff
@@ -193,8 +193,11 @@ pub use upgrade::*;
 ///
 /// The `L` is held and applied lazily during [`Upgradable::upgrade`].
 pub struct Operation<S, L = Identity> {
-    inner: S,
-    layer: L,
+    /// The inner [`Service`](tower::Service) representing the logic of the operation.
```

Comment:
These have become public in order to allow `Plugin` developers access to them.

Is this too severe? Maybe we should document that the average customer should use the `from_handler`/`from_service` methods.

Reply:
If we want customers to be able to write their own plugins, along the same lines of what we are doing here with the filtering, I don't see many other options.